docs/proposals/server-side-apply-field-managers.md

```diff
@@ -0,0 +1,181 @@
+
+## Summary
+
+ArgoCD supports server-side apply, but it uses the same field manager, `argocd-controller`, no matter what Application is issuing the sync. Setting a unique field manager per application would enable users to:
+
+- Manage only specific fields they care about on a shared resource
+- Avoid deleting or overwriting fields that are managed by other Applications
+
+## Motivation
+
+There exist situations where disparate Applications need to add or remove configuration from a shared Kubernetes resource. Server-side apply supports this behavior when different field managers are used.
+
+## Goals
+
+All following goals should be achieve in order to conclude this proposal:
+
+#### [G-1] Applications can apply changes to a shared resource without disturbing existing fields
+
+A common use case of server-side apply is the ability to manage only particular fields on a share Kubernetes resource, while leaving everything else in tact. This requires a unique field manager for each identity that shares the resource.
+
+#### [G-2] Applications that are destroyed only remove the fields they manage from shared resources
+
+A delete operation should undo only the additions or updates it made to a shared resource, unless that resource has `Prune=true`.
+
+## Non-Goals
+
+1. We don't intend to give users control of this field manager. Letting users change the field manager could lead to situations where fields get left behind on shared objects.
+
+## Proposal
+
+1. Each Application uses a field manager that is unique to itself and chosen deterministicly. It never changes for the lifetime of the Application.
+
+1. Change the removal behavior for shared resources. When a resource with a custom field manager is "removed", it instead removes only the fields managed by its field manager from the shared resource by sending an empty "fully specified intent" using server-side apply. You can fully delete a shared resource by setting `Prune=true` at the resource level. [Demo of this behavior](#removal-demo).
+
+1. Add documentation suggesting that users might want to consider changing the permissions on the ArgoCD role to disallow the `delete` and `update` verbs on shared resources. Server-side apply will always use `patch`, and removing `delete` and `update` helps prevent users from errantly wiping out changes made from other Applications.
+
+1. Add documentation that references the Kubernetes docs to show users how to properly define their Custom Resource Definition so that patches merge with expected results. For example, should a particular array on the resource be replaced or added to when patching?
+
+### Use cases
+
+The following use cases should be implemented:
+
+#### [UC-1]: As a user, I would like to manage specific fields on a Kubernetes object shared by other ArgoCD Applications.
+
+Change the Server-Side Apply field manager to be unique and deterministic for each Application.
```

Comment:
We need to add additional use-cases in this section to make sure the proposal is compatible with existing features.

Reply:
I added a couple more use cases. Let me know if you had some specific features in mind and I can add some more.